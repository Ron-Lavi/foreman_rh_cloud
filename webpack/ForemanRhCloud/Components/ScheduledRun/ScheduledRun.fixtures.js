export const props = {
  date: '2019-08-21T16:14:16.520+03:00',
  autoUploadEnabled: true,
};
