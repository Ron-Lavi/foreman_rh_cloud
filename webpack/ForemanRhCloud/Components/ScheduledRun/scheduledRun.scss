.scheduled_run {
  font-size: 14px;

  p {
    margin: 0 0 -10px 0;

    .fa {
      margin-right: 5px;
    }
  }
}
