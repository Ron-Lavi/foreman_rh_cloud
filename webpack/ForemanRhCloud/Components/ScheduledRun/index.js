import { connect } from 'react-redux';

import ScheduledRun from './ScheduledRun';
import { selectAutoUploadEnabled } from '../AccountList/AccountListSelectors';

// map state to props
const mapStateToProps = state => ({
  autoUploadEnabled: selectAutoUploadEnabled(state),
});

// export connected component
export default connect(mapStateToProps)(ScheduledRun);
