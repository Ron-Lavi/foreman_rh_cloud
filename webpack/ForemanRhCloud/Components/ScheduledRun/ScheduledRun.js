import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Grid } from 'patternfly-react';
import { FormattedRelative } from 'react-intl';
import { translate as __ } from 'foremanReact/common/I18n';
import './scheduledRun.scss';

const ScheduledRun = ({ date, autoUploadEnabled }) =>
  autoUploadEnabled && date ? (
    <Grid.Col sm={12} className="scheduled_run">
      <p>
        <Icon name="calendar" />
        {__('Next run: ')}
        <FormattedRelative value={date} />.
      </p>
    </Grid.Col>
  ) : null;

ScheduledRun.propTypes = {
  date: PropTypes.string,
  autoUploadEnabled: PropTypes.bool,
};

ScheduledRun.defaultProps = {
  date: null,
  autoUploadEnabled: true,
};

export default ScheduledRun;
