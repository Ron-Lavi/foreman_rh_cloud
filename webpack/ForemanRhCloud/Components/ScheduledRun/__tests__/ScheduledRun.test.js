import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import ScheduledRun from '../ScheduledRun';
import { props } from '../ScheduledRun.fixtures';

const fixtures = {
  'render with Props': props,
};

describe('ScheduledRun', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(ScheduledRun, fixtures));
});
