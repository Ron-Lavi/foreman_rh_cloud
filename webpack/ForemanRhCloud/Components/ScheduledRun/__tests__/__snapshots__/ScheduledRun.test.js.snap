// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduledRun rendering render with Props 1`] = `
<Col
  bsClass="col"
  className="scheduled_run"
  componentClass="div"
  sm={12}
>
  <p>
    <Icon
      name="calendar"
      type="fa"
    />
    Next run: 
    <FormattedRelative
      updateInterval={10000}
      value="2019-08-21T16:14:16.520+03:00"
    />
    .
  </p>
</Col>
`;
